dockers/docker-lldp-sv2/lldpmgrd

```diff
@@ -0,0 +1,193 @@
+
+class LldpManager(object):
+    """
+    Class which subscribes to notifications of changes in the PORT table of
+    the Redis State database and updates LLDP configuration accordingly for
+    that port by calling lldpcli.
+    Attributes:
+        state_db: Handle to Redis State database via swsscommon lib
+        config_db: Handle to Redis Config database via swsscommon lib
+    """
+    REDIS_HOSTNAME = "localhost"
+    REDIS_PORT = 6379
+    REDIS_TIMEOUT_USECS = 0
+
+    def __init__(self):
+        # Open a handle to the State database
+        self.state_db = swsscommon.DBConnector(swsscommon.STATE_DB,
+                                               self.REDIS_HOSTNAME,
+                                               self.REDIS_PORT,
+                                               self.REDIS_TIMEOUT_USECS)
+
+        # Open a handle to the Config database
+        self.config_db = swsscommon.DBConnector(swsscommon.CONFIG_DB,
+                                                self.REDIS_HOSTNAME,
+                                                self.REDIS_PORT,
+                                                self.REDIS_TIMEOUT_USECS)
+
+    def update_lldp_config_for_port(self, port_name):
+        """
+        For port `port_name`, look up the neighboring device's hostname and
+        corresponding port alias in the Config database, then form the
+        appropriate lldpcli configuration command and run it.
+        """
+        CMD_FMT = "lldpcli configure ports {0} lldp portidsubtype local {1} description {2}:{3}"
+        TABLE_SEPARATOR = "|"
+
+        # Retrieve all entires from the Device Neighbor table
+        device_neighbor_table = swsscommon.Table(self.config_db, swsscommon.CFG_DEVICE_NEIGHBOR_TABLE_NAME, TABLE_SEPARATOR)
+        (status, fvp) = device_neighbor_table.get(port_name)
+        if not status:
+            log_info("Port '{}' not found in {} table in Config DB. Not updating LLDP config for this port.".format(port_name, swsscommon.CFG_DEVICE_NEIGHBOR_TABLE_NAME))
+            return
+
+        # Convert list of tuples to a dictionary
+        device_neighbor_table_dict = dict(fvp)
+
+        neighbor_hostname = device_neighbor_table_dict["name"]
+        if not neighbor_hostname:
+            log_error("Failed to obtain neighbor hostname for port '{}'".format(port_name))
+            return
+
+        neighbor_portname = device_neighbor_table_dict["port"]
+        if not neighbor_hostname:
+            log_error("Failed to obtain neighbor port name for port '{}'".format(port_name))
+            return
+
+        # Retrieve all entires from the Port table
+        port_table = swsscommon.Table(self.config_db, swsscommon.CFG_PORT_TABLE_NAME, TABLE_SEPARATOR)
+        (status, fvp) = port_table.get(port_name)
+        if not status:
+            log_error("Port '{}' not found in {} table in Config DB.".format(port_name, swsscommon.CFG_PORT_TABLE_NAME))
```

Comment:
to be complete, we should also listen to PORT_TABLE_NAME change, same reason as above.